plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/pdtracker/PlantDescriptionTracker.java

```diff
@@ -0,0 +1,336 @@
+package eu.arrowhead.core.plantdescriptionengine.pdtracker;
+
+import eu.arrowhead.core.plantdescriptionengine.pdtracker.backingstore.PdStore;
+import eu.arrowhead.core.plantdescriptionengine.pdtracker.backingstore.PdStoreException;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.Connection;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.PdeSystem;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.PlantDescriptionEntry;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.PlantDescriptionEntryDto;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.PlantDescriptionEntryListDto;
+import eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.Port;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Object used to keep track of Plant Description entries. It keeps a reference
+ * to a {@link eu.arrowhead.core.plantdescriptionengine.pdtracker.backingstore},
+ * which is used to store Plant Description Entries in some permanent storage
+ * (e.g. to file or a database).
+ */
+public class PlantDescriptionTracker {
+
+    // List of instances that need to be informed of any changes to Plant
+    // Description Entries.
+    final List<PlantDescriptionUpdateListener> listeners = new ArrayList<>();
+
+    // ID-to-entry mapping
+    private final Map<Integer, PlantDescriptionEntryDto> entries = new ConcurrentHashMap<>();
+
+    // Non-volatile storage for entries:
+    private final PdStore backingStore;
+
+    // Integer for storing the next plant description entry ID to be used:
+    private final AtomicInteger nextId = new AtomicInteger();
+
+    /**
+     * Class constructor.
+     *
+     * @param backingStore Non-volatile storage for entries.
+     * @throws PdStoreException If backing store operations fail.
+     */
+    public PlantDescriptionTracker(final PdStore backingStore) throws PdStoreException {
+        Objects.requireNonNull(backingStore, "Expected backing store");
+        this.backingStore = backingStore;
+
+        // Read entries from non-volatile storage and
+        // calculate the next free Plant Description Entry ID:
+        int maxId = -1;
+        for (final PlantDescriptionEntryDto entry : backingStore.readEntries()) {
+            maxId = Math.max(maxId, entry.id());
+            entries.put(entry.id(), entry);
+        }
+        nextId.set(Math.round(maxId + 1));
```

Comment:
why do you need round here? Every number is int.

Reply:
ID:s used to be floats. Fixed now.